docs/design/ocs-to-ocs-api-server.md

```diff
@@ -0,0 +1,201 @@
+  rpc GetStorageConfig(StorageConfigRequest)
+  returns (StorageConfigResponse) {}
+  // UpdateCapacity RPC call to increase or decrease the storage pool size
+  rpc UpdateCapacity(UpdateCapacityRequest)
+  returns (UpdateCapacityResponse) {}
+  // OffboardConsumer RPC call to delete StorageConsumer CR on the storage provider cluster.
+  rpc OffboardConsumer (OffboardConsumerRequest)
+  returns (OffboardConsumerResponse) {}
+}
+```
+
+#### OnboardConsumer
+- Check if any existing `StorageConsumer` CR with same token is available and return its encrypted UUID
+- If no existing `StorageConsumer` CR found:
+    - Create `StorageConsumer` CR after validating the token.
+    - Return the encrypted UUID of the `StorageConsumer` CR
+
+```protobuf
+// OnBoardConsumerRequest holds the required information to validate the consumer and create StorageConsumer
+// resource on the StorageProvider cluster
+message OnBoardConsumerRequest{
+    // token provided by the provider cluster admin to authenticate the consumer
+    string token =1;
+    // capacity is a valid k8s resource quantity
+    string capacity = 2;
+}
+
+// OnBoardConsumerResponse holds the response for OnBoardConsumer API request
+message OnBoardConsumerResponse{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+}
+
+```
+
+##### Response Scheme
+
+
+| Condition | gRPC Code | Description | Recovery |
+| -------- | -------- | -------- | -------- |
+| Success    | 0  OK     | Request Successful. StorageConsumer CR is created    | N/A |
+| Invalid ID | 3 INVALID_ARGUMENT | Invalid request token or token expired | Contact the Provider Cluster Admin to verify the token
+| Error | 13 INTERNAL | Request failed due to some error in provider cluster. For example: failed to create the StorageConsumer CR| Contact Provider Cluster Admin
+
+
+#### GetStorageConfig
+- Read the `status.state` of the StorageConsumer CR:
+  - if `status` is `Failed`, return error.
+  - if `status` is `Configuring`, return `UNAVAILABLE`.
+  - if `status` is `Ready`, fetch the `blockPoolName` and the`cephUser` details from the CR status. Generate and return json connection details.
+
+**Note**: When `UNAVAILABLE` status is returned, the consumer controller should requeue (with interval) the reconciler to call the `GetProviderConfig` API again to check if the rook resources are available in the `StorageConsumer` CR status.
+
+```protobuf
+// StorageConfigRequest holds the information required generate the json config for connecting to storage provider cluster
+message StorageConfigRequest{
+   // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID =1;
+}
+
+// StorageConfigResponse holds the response for the GetStorageConfig API request
+message StorageConfigResponse{
+    // data contains the json blob to be used by the consumer cluster to connect with the provider cluster
+    google.protobuf.Struct data = 1;
```

Comment:
How about calling it blob?

Reply:
It is missing  the type: bytes

Reply:
replaced `google.protobuf.Struct` with bytes.  I don't think `blob` might be good idea. It could be equally confusing as `data`, if not less.